### PR TITLE
Fix dygraph unique name bug

### PR DESCRIPTION
**Pull request by sneaxiy**
`unique_name.ids` increases as batch increases, which make memory leak in dygraph mode since dygraph mode creates new variables across batches. This bug makes ptb model run slower as batch increases. This PR fixes this bug.

### CODE REVIEW

File: python/paddle/fluid/unique_name.py [hunk 1]
```diff
@@ -58,6 +58,14 @@ def generate(key):
     return generator(key)
 
 
+def generate_tmp(key):
```
Comment (JiabinYang) [inline python/paddle/fluid/unique_name.py]: Add Comment to explain why we need generate_tmp

**Review by chengduoZH — DISMISSED** [review]
LGTM

**Review by JiabinYang — APPROVED** [review]
LGTM

File: python/paddle/fluid/unique_name.py (hunk 1 above)
Comment (sneaxiy) [reply python/paddle/fluid/unique_name.py]: Done.

**Pull request merged**